### PR TITLE
fix(cli): reduce flicker in spinner render function

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: cli/spinner.ts [hunk 1]
```diff
@@ -136,13 +136,16 @@ export class Spinner {
...
         noColor
           ? this.#spinner[i] + " " + this.message
           : color + this.#spinner[i] + COLOR_RESET + " " + this.message,
       );
-      Deno.stdout.writeSync(frame);
+      // call writeSync once to reduce flickering
+      const writeData = new Uint8Array(LINE_CLEAR.length + frame.length);
+      writeData.set(LINE_CLEAR);
+      writeData.set(frame, LINE_CLEAR.length);
+      Deno.stdout.writeSync(writeData);
```
Comment (iuioiua) [inline cli/spinner.ts:148]: An alternative using [`concat()`](https://jsr.io/@std/bytes@1.0.0-rc.3/doc/concat/~/concat):
```suggestion
      Deno.stdout.writeSync(concat([LINE_CLEAR, frame]));
```

**Review by kt3k — APPROVED** [review]
LGTM

File: cli/spinner.ts (hunk 1 above)
Comment (dsherret) [reply cli/spinner.ts:148]: Oh, I missed that this was in a separate package. I'm not sure it's worth causing another package load to occur when importing this module.

**Review by iuioiua — APPROVED** [review]
It's no big deal omitting `concat()` for just two `Uint8Array`s. LGTM.

**Pull request merged**